File: frontend/src/app/core/models/watches/watch-update-request.dto.ts
import { WatchCondition } from "./enums/watch-condition";
import { WatchGender } from "./enums/watch-gender";
import { WatchMovementType } from "./enums/watch-movement-type";
import { WatchType } from "./enums/watch-type";

export interface WatchUpdateRequestDTO
{
    manufacturer: string,
    model: string,
    referenceNumber: string,
    movement: string,
    description: string,
    yearOfProduction: number,
    pricePerDay: number,
    condition: WatchCondition,
    gender: WatchGender,
    movementType: WatchMovementType,
    watchType: WatchType
}

File: backend/src/main/java/com/yourwatchrental/watchrental/watch/dto/request/WatchUpdateRequestDTO.java
package com.yourwatchrental.watchrental.watch.dto.request;

import com.yourwatchrental.watchrental.watch.enums.Condition;
import com.yourwatchrental.watchrental.watch.enums.Gender;
import com.yourwatchrental.watchrental.watch.enums.MovementType;
import com.yourwatchrental.watchrental.watch.enums.WatchType;
import jakarta.validation.constraints.PositiveOrZero;

import java.math.BigDecimal;

public record WatchUpdateRequestDTO(
        String manufacturer,

        String model,

        String referenceNumber,

        String movement,

        String description,

        @PositiveOrZero(message = "Year of production needs to be accurate")
        int yearOfProduction,

        @PositiveOrZero(message = "Price per day needs to be accurate")
        BigDecimal pricePerDay,

        Condition condition,

        Gender gender,

        MovementType movementType,

        WatchType watchType
) {
}
